examples/save_plot/src/main.rs

```diff
@@ -0,0 +1,148 @@
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use std::path::PathBuf;
+
+use eframe::egui;
+use eframe::egui::plot::{Legend, Line, Plot, PlotPoints};
+use eframe::egui::ColorImage;
+
+fn main() -> Result<(), eframe::Error> {
+    // Log to stdout (if you run with `RUST_LOG=debug`).
+    tracing_subscriber::fmt::init();
+
+    let options = eframe::NativeOptions {
+        initial_window_size: Some(egui::vec2(350.0, 400.0)),
+        ..Default::default()
+    };
+    eframe::run_native(
+        "My egui App with a plot",
+        options,
+        Box::new(|_cc| Box::new(MyApp::default())),
+    )
+}
+
+struct MyApp {
+    picked_path_plot: PathBuf,
+    plot_location: [f32; 4],
+    save_plot: bool,
+    screenshot: Option<ColorImage>,
+}
+
+impl Default for MyApp {
+    fn default() -> Self {
+        Self {
+            picked_path_plot: PathBuf::default(),
+            plot_location: [0.0; 4],
+            save_plot: false,
+            screenshot: None,
+        }
+    }
+}
+
+impl eframe::App for MyApp {
+    fn update(&mut self, ctx: &egui::Context, frame: &mut eframe::Frame) {
+        egui::CentralPanel::default().show(ctx, |ui| {
+            let height = 200.0;
+            let border_x = 11.0;
+            let border_y = 18.0;
+            let width = 300.0;
+
+            let window_width = ui.available_size().x;
+            let window_height = ui.available_size().y;
+
+            ui.heading("My egui Application");
+
+            // add some whitespace in y direction
+            ui.add_space(border_y);
+
+            if ui.button("Save Plot").clicked() {
+                if let Some(mut path) = rfd::FileDialog::new().save_file() {
+                    path.set_extension("png");
+                    frame.request_screenshot();
+                    self.save_plot = true;
+                    self.picked_path_plot = path;
+                }
+            }
```

Comment:
You can simplify the state here - no need for `save_plot` nor `picked_path_plot` if you instead use the logic:


```rs
if ui.button("Save Plot").clicked() {
    frame.request_screenshot();
}

```

and then late

```rs
if let Some(screenshot) = self.screenshot.take() {
    if let Some(mut path) = rfd::FileDialog::new().save_file() {
        // save to path
    }
}
```

Reply:
done